log-manager/src/main/java/io/airlift/log/LogJmxModule.java

```diff
@@ -34,12 +36,28 @@ public void configure(Binder binder)
 
         binder.bind(LoggingMBean.class).in(Scopes.SINGLETON);
         newExporter(binder).export(LoggingMBean.class).as("io.airlift.log:name=Logging");
+
+        // Eager to ensure JMX registration
+        binder.bind(Logging.class).toProvider(LoggingProvider.class).asEagerSingleton();
     }
 
-    @Provides
-    @Singleton
-    public Logging getLogging()
+    public static class LoggingProvider
+            implements Provider<Logging>
     {
-        return Logging.initialize();
+        private final MBeanExporter exporter;
+
+        @Inject
+        public LoggingProvider(MBeanExporter exporter)
+        {
+            this.exporter = requireNonNull(exporter, "exporter is null");
+        }
+
+        @Override
+        public Logging get()
+        {
+            Logging logging = Logging.initialize();
+            logging.registerJmxExports(exporter);
```

Comment:
I'm pretty sure we need to unexport when the system is shutdown.  Todo this, I suggest reverting this code back to a signleton provider method and adding this:
```java
@Override
public void configure(Binder binder)
{
    binder.disableCircularProxies();

    binder.bind(LoggingMBean.class).in(Scopes.SINGLETON);
    newExporter(binder).export(LoggingMBean.class).as("io.airlift.log:name=Logging");
    binder.bind(LogExporter.class).asEagerSingleton();
}

@Provides
@Singleton
public Logging getLogging()
{
    return Logging.initialize();
}

public static class LogExporter
{

    private final Logging logging;
    private final MBeanExporter exporter;

    @Inject
    public LogExporter(Logging logging, MBeanExporter exporter)
    {
        this.logging = requireNonNull(logging, "logging is null");
        this.exporter = requireNonNull(exporter, "exporter is null");
        logging.export(exporter);
    }

    @PreDestroy
    public void destroy()
    {
        logging.unexport(exporter);
    }
}
```